contracts/router/interfaces/IRouterV2.sol

```diff
@@ -0,0 +1,113 @@
+    function idToModule(bytes32 moduleId) external view returns (address bridgeModule);
+
+    /// @notice Gets the module ID associated with the given bridge module address
+    /// @param bridgeModule Bridge module address
+    /// @return moduleId Unique bridge module ID
+    function moduleToId(address bridgeModule) external view returns (bytes32 moduleId);
+
+    /// @notice Gets the list of all bridge tokens (and their symbols), such that destination swap
+    /// from a bridge token to `tokenOut` is possible.
+    /// @param tokenOut  Token address to swap to on destination chain
+    /// @return destTokens List of structs with following information:
+    ///                  - symbol: unique token ID consistent among all chains
+    ///                  - token: bridge token address
+    function getDestinationBridgeTokens(address tokenOut) external view returns (BridgeToken[] memory destTokens);
+
+    /// @notice Gets the list of all bridge tokens (and their symbols), such that origin swap
+    /// from `tokenIn` to a bridge token is possible.
+    /// @param tokenIn  Token address to swap from on origin chain
+    /// @return originTokens List of structs with following information:
+    ///                  - symbol: unique token ID consistent among all chains
+    ///                  - token: bridge token address
+    function getOriginBridgeTokens(address tokenIn) external view returns (BridgeToken[] memory originTokens);
+
+    /// @notice Gets the list of all tokens that can be swapped into bridge tokens on this chain.
+    /// @return supportedTokens Supported token addresses that can be swapped for bridge tokens
+    function getSupportedTokens() external view returns (address[] memory supportedTokens);
+
+    /// @notice Finds the best path between every supported bridge token from the given list and `tokenOut`,
+    /// treating the swap as "destination swap", limiting possible actions to those available for every bridge token.
+    /// @dev Will NOT revert if any of the tokens are not supported, instead will return an empty query for that symbol.
+    /// Note: it is NOT possible to form a SwapQuery off-chain using alternative SwapAdapter for the destination swap.
+    /// For the time being, only swaps through the Synapse-supported pools are available on destination chain.
+    /// @param requests  List of structs with following information:
+    ///                 - symbol: unique token ID consistent among all chains
+    ///                 - amountIn: amount of bridge token to start with, before the bridge fee is applied
+    /// @param tokenOut  Token user wants to receive on destination chain
+    /// @return destQueries  List of structs that could be used as `destQuery` in SynapseRouter.
+    ///                      minAmountOut and deadline fields will need to be adjusted based on the user settings.
+    function getDestinationAmountOut(DestRequest[] memory requests, address tokenOut)
```

Comment:
If you feel that this and `getOriginAmountOut` could have better names - feel free to shoot the suggestions. Could also stay as is though.